### PR TITLE
Match __repr__ to attribute

### PR DESCRIPTION
**Pull request by DanielLenz**
Rename Error -> Uncertainty and Units -> Unit in `__repr__` to match the attributes.
Requested in https://github.com/astropy/astropy/issues/4145
Had to re-do the pull request because I messed up the branches.

### CODE REVIEW

**Comment by dlenz-bonn** [comment]
Had to re-do the pull request because I messed up the branches.

**Comment by pllim** [comment]
Note: Old PR was #4146. 

:+1: It seems uncontraversial. I would lean towards a change log, because why not?

**Comment by mhvk** [comment]
I think indeed it is best to err in the direction of too much information: @dienz-bonn: could you add an entry to the file `CHANGES.rst`? I think it would go under the header of `Bug fixes` (for 1.1), since really a `repr` is meant to be something that one can reproduce an instance with, and it would seema bug for it not to have the right names.

In the commit message for that change, you can add "[skip ci]", so that travis does not run again, and then this can be merged.  Thanks!

**Comment by mhvk** [comment]
Sorry for the misspelled id, @dlenz-bonn

**Comment by DanielLenz** [comment]
Thanks for the help, I added the changes to the change log. Note that I changed my username as well, but the commits should appear with the old id.

**Comment by mhvk** [comment]
Great, thanks!  Merging...

**Comment by embray** [comment]
Why should this be limited to v1.1.0?  This change can be backported to the v1.0.x branch.

File: astropy/constants/constant.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def __new__(cls, abbrev, name, value, unit, uncertainty, reference,
         return inst
 
     def __repr__(self):
-        return ('<Constant name={0!r} value={1} error={2} units={3!r} '
+        return ('<Constant name={0!r} value={1} Uncertainty={2} unit={3!r} '
```
Comment (embray) [inline astropy/constants/constant.py]: This should have been "uncertainty" in lowercase.

**Comment by embray** [comment]
I'm putting this in 1.0.5.

**Pull request merged**